File: user-service/src/main/java/socialapp/userservice/service/UserService.java
package socialapp.userservice.service;

import org.springframework.data.elasticsearch.core.SearchHits;
import org.springframework.web.multipart.MultipartFile;
import socialapp.userservice.model.dto.*;
import socialapp.userservice.model.entity.User;

import java.util.List;
import java.util.Set;

public interface UserService {

    User register(RegistrationDto registrationDto);

    void delete(Long id);

    Set<SuggestionResponse> fetchSuggestions(String query);

    IsExistsResponse isExists(String email);

    void uploadAvatar(MultipartFile file, Long id);

    void update(UserUpdateDto userUpdateDto,Long id);

    SearchHits<User> find(UserSearchCriteria userSearchCriteria, int page, int pageSize);

    User findById(Long id);

}

File: kt-user-service/src/main/kotlin/socialapp/ktuserservice/service/UserService.kt
package socialapp.ktuserservice.service

import org.springframework.data.elasticsearch.core.SearchHits
import org.springframework.web.multipart.MultipartFile
import socialapp.ktuserservice.model.dto.IsExistsResponse
import socialapp.ktuserservice.model.dto.RegistrationDto
import socialapp.ktuserservice.model.dto.UserSearchCriteria
import socialapp.ktuserservice.model.dto.UserUpdateDto
import socialapp.ktuserservice.model.entity.User

interface UserService {

    fun register(registrationDto: RegistrationDto): User

    fun delete(id: Long)

//    fun fetchSuggestions(query: String): Set<SuggestionResponse>

    fun isExists(email: String): IsExistsResponse

    fun uploadAvatar(file: MultipartFile, id: Long)

    fun update(userUpdateDto: UserUpdateDto, id: Long)

    fun find(userSearchCriteria: UserSearchCriteria, page: Int, pageSize: Int): SearchHits<User>

    fun findById(id: Long): User
}